cpp/src/arrow/filesystem/s3_internal.h

```diff
@@ -88,6 +88,54 @@ inline bool IsAlreadyExists(const Aws::Client::AWSError<Aws::S3::S3Errors>& erro
           error_type == Aws::S3::S3Errors::BUCKET_ALREADY_OWNED_BY_YOU);
 }
 
+std::string S3ErrorToString(Aws::S3::S3Errors error_type) {
```

Comment:
Should make this `inline` to avoid potential multiple definition errors.

Reply:
👍